File: hyeonsook95/leetcode/20.py
class Solution:
    def isValid(self, s: str) -> bool:
        parentheses = {")": "(", "]": "[", "}": "{"}
        temp = []

        for char in s:
            if char in parentheses.values():
                temp.append(char)
            elif char in parentheses.keys():
                if temp and temp[-1] == parentheses[char]:
                    temp.pop()
                else:
                    return False

        if temp:
            return False
        return True

File: hyeonsook95/leetcode/20.ts
function isValid(s) {
    var parentheses = { ")": "(", "]": "[", "}": "{" };
    var temp = [];

    for (var char of s) {
        if ("([{".indexOf(char) > -1) {
            temp.push(char);
        }
        else {
            if (temp.length === 0) {
                return false;
            }
            var top = temp.pop();
            if (top !== parentheses[char]) {
                return false;
            }
        }
    }

    if (temp.length !== 0) {
        return false;
    }
    return true;
};

console.log(isValid(')'));